example/pythonlib/linting/1-ruff-format/build.mill

```diff
@@ -0,0 +1,86 @@
+// First, make your module extend `pythonlib.RuffModule`.
+
+package build
+import mill._, pythonlib._
+
+object `package` extends RootModule with PythonModule with RuffModule
+
+// You can reformat your project's code by running the `ruffFormat` task.
+
+/** Usage
+
+> cat src/main.py # initial poorly formatted source code
+from typing import Self
+class IntWrapper:
+   def __init__(self, x:int):
+     self.x    =x
+   def plus(self, w:Self) ->   Self:
+      return     IntWrapper(self.x + w.x)
+print(IntWrapper(2).plus(IntWrapper(3)).x)
+...
+
+> mill ruffFormat --diff # you can also pass in extra arguments understood by `ruff format`
+error: ...
+error: @@ -1,7 +1,12 @@
+error:  from typing import Self
+error: +
+error: +
+error:  class IntWrapper:
+error: -   def __init__(self, x:int):
+error: -     self.x    =x
+error: -   def plus(self, w:Self) ->   Self:
+error: -      return     IntWrapper(self.x + w.x)
+error: +    def __init__(self, x: int):
+error: +        self.x = x
+error: +
+error: +    def plus(self, w: Self) -> Self:
+error: +        return IntWrapper(self.x + w.x)
+error: +
+error: +
+error:  print(IntWrapper(2).plus(IntWrapper(3)).x)
+error: ...
+error: 1 file would be reformatted
+
+> mill ruffFormat
+...1 file reformatted
```

Comment:
FYI, I wanted to first show how to call `ruffFormat` and *then* how to add arguments, e.g. `ruffFormat --diff`. Unfortunately however `ruffFormat` changes the original file, so there isn't really a nice way to do that.

Reply:
We can use `sed` to mangle files as necessary in the examples (or `sed -i.bak` for cross platform support, see other examples). But I think this example is already sufficient TBH, no need to exercise every code path in the example, just the main ones. The others like passing in arguments can be done in just textual documentation